core/src/main/java/org/apache/spark/shuffle/sort/io/LocalDiskShuffleMapOutputWriter.java

```diff
@@ -124,6 +137,8 @@ private void cleanUp() throws IOException {
 
   private void initStream() throws IOException {
     if (outputFileStream == null) {
+      // This file needs to opened in append mode in order to work around a Linux kernel bug that
+      // affects transferTo; see SPARK-3948 for more details.
```

Comment:
comment is actually more appropriate in `initChannel()` (transferTo is only used w/ the channel).

btw, why does `initChannel` set the member variable `outputFileStream`?  Couldn't it just set

`outputChannel = new FileOutputStream(outputTempFile, true).getChannel();`?